tests/python/pants_test/backend/jvm/tasks/jvm_compile/java/test_zinc_compile_integration.py

```diff
@@ -431,60 +431,34 @@ def main(args: Array[String]) {
             path = os.path.join(compile_dir, path_suffix)
             self.assertTrue(os.path.exists(path), "Want path {} to exist".format(path))
 
-  def test_hermetic_binary_with_3rdparty_dependencies_ivy(self):
-    config = {
-      'resolve.ivy': {'capture_snapshots': True},
-      'compile.zinc': {
-        'execution_strategy': 'hermetic',
-        'use_classpath_jars': False,
-        'incremental': False,
-      },
-      'resolver': {
-        'resolver': 'ivy',
-      }
-    }
-
-    with self.temporary_workdir() as workdir:
-      with self.temporary_file_content("readme.txt", b"yo"):
-        pants_run = self.run_pants_with_workdir(
-          [
-            'run',
-            'testprojects/src/java/org/pantsbuild/testproject/cwdexample',
-          ],
-          workdir,
-          config,
-        )
-        self.assert_success(pants_run)
-        self.assertIn(
-          'Found readme.txt',
-          pants_run.stdout_data,
-        )
-
-  def test_hermetic_binary_with_3rdparty_dependencies_coursier(self):
-    config = {
-      'resolve.coursier': {'capture_snapshots': True},
-      'compile.zinc': {
-        'execution_strategy': 'hermetic',
-        'use_classpath_jars': False,
-        'incremental': False,
-      },
-      'resolver': {
-        'resolver': 'coursier',
-      }
-    }
+  def test_hermetic_binary_with_3rdparty_dependencies(self):
```

Comment:
It would be really incredibly super cool if we could simply move the `_for_all_execution_environments` decorator at https://github.com/pantsbuild/pants/blob/bf7263d00ed6f78f319cae03cc9cfe0c82d4aee4/tests/python/pants_test/backend/jvm/tasks/jvm_compile/rsc/test_rsc_compile_integration.py#L16 into this file (or a `util/` subdir or something) and just add `use_classpath_jars` into it. Thinking about it now, that's almost definitely something we'd want to *generate* tests for, since each individual test right now takes a very long time to run. If we could make a TODO + issue link that'd be pretty easy to get to this week or later next.

Reply:
See #7824: I think that following this week, we should try to collapse at least one permutation out.